src/Health/HealthDAO.ts

```diff
@@ -41,6 +43,18 @@ type updateNetworkInfo = (x: NetworkInfo) => Promise<UpdateWriteOpResult>
 
 type updateIPFSInfo = (x: IPFSInfo) => Promise<UpdateWriteOpResult>
 
+type insertIPFSFailure = (x: IPFSHashFailure) => Promise<UpdateWriteOpResult>
```

Comment:
I would prefer DOA's don't return mongo specific results, that way they hide the implementation details of the database. Return void if the user doesn't need anything from the result, if the user does need a result return a curated result that does not contain mongo details.

same goes for all the other functions below. ( looks like I missed one up above, don't touch that function in this pr refactor it in a separate pr )

Reply:
Ok I updated them. Look better now? I opened another PR that updates the others as well.